bakery/store.go

```diff
@@ -0,0 +1,81 @@
+package bakery
+
+import (
+	"sync"
+
+	"golang.org/x/net/context"
+	"gopkg.in/macaroon.v2-unstable"
+)
+
+// RootKeyStore defines store for macaroon root keys.
+type RootKeyStore interface {
+	// Get returns the root key for the given id.
+	// If the item is not there, it returns ErrNotFound.
+	Get(ctxt context.Context, id []byte) ([]byte, error)
+
+	// RootKey returns the root key to be used for making a new
+	// macaroon, and an id that can be used to look it up later with
+	// the Get method.
+	//
+	// Note that the root keys should remain available for as long
+	// as the macaroons using them are valid.
+	//
+	// Note that there is no need for it to return a new root key
+	// for every call - keys may be reused, although some key
+	// cycling is over time is advisable.
+	RootKey(ctxt context.Context) (rootKey []byte, id []byte, err error)
+}
+
+// NewMemRootKeyStore returns an implementation of
+// Store that generates a single key and always
+// returns that from RootKey. The same id ("0") is always
+// used.
+func NewMemRootKeyStore() RootKeyStore {
+	return new(memRootKeyStore)
+}
+
+type memRootKeyStore struct {
+	mu  sync.Mutex
```

Comment:
could use sync.RWMutex..  Get method reads the key and could use RLock and RUnlock..  or sync/atomic Value